kubernetes/adapter.go

```diff
@@ -217,7 +217,27 @@ func (a *Adapter) parseAnnotations(annotations map[string]string) *Ingress {
 		sslPolicy = a.ingressDefaultSSLPolicy
 	}
 
-	loadBalancerType := getAnnotationsString(annotations, ingressLoadBalancerTypeAnnotation, a.ingressDefaultLoadBalancerType)
+	loadBalancerType, hasLB := annotations[ingressLoadBalancerTypeAnnotation]
+	if !hasLB {
+		loadBalancerType = a.ingressDefaultLoadBalancerType
+	}
+
+	securityGroup, hasSG := annotations[ingressSecurityGroupAnnotation]
+	if !hasSG {
+		securityGroup = a.ingressDefaultSecurityGroup
+	}
+
+	wafWebAclId, hasWAF := annotations[ingressWAFWebACLIDAnnotation]
+
+	if (loadBalancerType == loadBalancerTypeNLB) && (hasSG || hasWAF) {
+		if hasLB {
+			log.Warn("Security Group and WAF are not supported by NLB (configured by annotation)")
```

Comment:
seems more Info level to me. The user decides that she wants to enforce NLB. It might be useful to have hasSG or hasWAF printed.